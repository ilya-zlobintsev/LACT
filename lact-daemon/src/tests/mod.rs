use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
