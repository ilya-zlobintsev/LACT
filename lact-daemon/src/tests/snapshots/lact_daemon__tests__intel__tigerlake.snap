---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "table": {
      "type": "intel",
      "value": {
        "gt_freq": [
          1100,
          1300
        ],
        "rp0_freq": 1300,
        "rpe_freq": 600,
        "rpn_freq": 100
      }
    }
  },
  "info": {
    "driver": "xe",
    "drm_info": {
      "memory_info": {
        "cpu_accessible_total": 0,
        "cpu_accessible_used": 0,
        "resizeable_bar": true
      },
      "vram_clock_ratio": 1.0
    },
    "flags": [],
    "link_info": {},
    "pci_info": {
      "device_pci_info": {
        "model": "TigerLake-LP GT2 [Iris Xe Graphics]",
        "model_id": "9A49",
        "vendor": "Intel Corporation",
        "vendor_id": "8086"
      },
      "subsystem_pci_info": {
        "model_id": "3F19",
        "vendor": "Lenovo",
        "vendor_id": "17AA"
      }
    },
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "TigerLake-LP GT2 [Iris Xe Graphics]",
      "model_id": "9A49",
      "vendor": "Intel Corporation",
      "vendor_id": "8086"
    },
    "subsystem_pci_info": {
      "model_id": "3F19",
      "vendor": "Lenovo",
      "vendor_id": "17AA"
    }
  },
  "power_profile_modes": null,
  "power_states": {
    "core": [
      {
        "enabled": true,
        "value": 100
      },
      {
        "enabled": true,
        "value": 600
      },
      {
        "enabled": true,
        "value": 1300
      }
    ],
    "vram": []
  },
  "stats": {
    "clockspeed": {
      "current_gfxclk": 1100,
      "gpu_clockspeed": 1100
    },
    "fan": {
      "control_enabled": false,
      "pmfw_info": {}
    },
    "power": {
      "cap_min": 0.0
    },
    "temps": {},
    "throttle_info": {},
    "voltage": {},
    "vram": {}
  }
}
